core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/util/CryptoUtil.java

```diff
@@ -170,6 +170,58 @@ public RegistryService getRegistryService() {
         return encrypt(plainTextBytes, System.getProperty(CIPHER_TRANSFORMATION_SYSTEM_PROPERTY), true);
     }
 
+    /**
+     * Encrypt a given plain text with the provided algorithm and internal crypto provider type.
+     *
+     * @param plainTextBytes                plain text bytes that need to be encrypted.
+     * @param algorithm                     The encryption algorithm.
+     * @param returnSelfContainedCipherText Create self-contained cipher text if true, return simple encrypted ciphertext otherwise.
+     * @param internalCryptoProviderType    The {@link org.wso2.carbon.crypto.api.InternalCryptoProvider} type.
+     * @return The cipher text bytes.
+     * @throws CryptoException
+     */
+    public byte[] encrypt(byte[] plainTextBytes, String algorithm, boolean returnSelfContainedCipherText,
+                          String internalCryptoProviderType)
+            throws CryptoException {
+
+        failIfEncryptDecryptInputsAreInvalid(plainTextBytes, algorithm, internalCryptoProviderType);
+
+        byte[] encryptedKey;
+        try {
+
+            CryptoService cryptoService = CarbonCoreDataHolder.getInstance().getCryptoService();
+
+            if (cryptoService == null) {
+                throw new CryptoException("A crypto service implementation has not been registered.");
+            }
+            if (plainTextBytes.length == 0) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Plaintext is empty. An empty array will be used as the ciphertext bytes.");
+                }
+                encryptedKey = StringUtils.EMPTY.getBytes();
+            } else {
+                if (log.isDebugEnabled()) {
+                    if (returnSelfContainedCipherText) {
+                        log.debug(String.format(
+                                "Encrypting data in self contained format, with algorithm: '%s' and internal crypto " +
+                                        "provider: '%s'", algorithm, internalCryptoProviderType));
+                    } else {
+                        log.debug(String.format(
+                                "Encrypting data with algorithm: '%s' and internal crypto " +
+                                        "provider: '%s'", algorithm, internalCryptoProviderType));
+                    }
+                }
+                encryptedKey = cryptoService
+                        .encrypt(plainTextBytes, algorithm, CRYPTO_API_PROVIDER_BC, returnSelfContainedCipherText,
+                                internalCryptoProviderType);
+            }
+
+        } catch (Exception e) {
```

Comment:
Can't we catch the specific exception here?

Reply:
Fixed with 7157b37d5c65be094165fc2ee48137ca915b81c3